File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionPaymentRepositoryImpl.java
package com.eager.questioncloud.core.domain.payment.infrastructure.repository;

import com.eager.questioncloud.core.domain.payment.infrastructure.entity.QuestionPaymentEntity;
import com.eager.questioncloud.core.domain.payment.model.QuestionPayment;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class QuestionPaymentRepositoryImpl implements QuestionPaymentRepository {
    private final QuestionPaymentJpaRepository questionPaymentJpaRepository;

    @Override
    public QuestionPayment save(QuestionPayment questionPayment) {
        QuestionPaymentEntity entity = questionPaymentJpaRepository.save(QuestionPaymentEntity.from(questionPayment));
        questionPayment.success(entity.getId());
        return questionPayment;
    }

    @Override
    public int countByUserId(Long userId) {
        return questionPaymentJpaRepository.countByUserId(userId);
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionPaymentRepositoryImpl.kt
package com.eager.questioncloud.core.domain.payment.infrastructure.repository

import com.eager.questioncloud.core.domain.payment.infrastructure.entity.QuestionPaymentEntity
import com.eager.questioncloud.core.domain.payment.model.QuestionPayment
import org.springframework.stereotype.Repository

@Repository
class QuestionPaymentRepositoryImpl(
    private val questionPaymentJpaRepository: QuestionPaymentJpaRepository
) : QuestionPaymentRepository {

    override fun save(questionPayment: QuestionPayment): QuestionPayment {
        questionPaymentJpaRepository.save(QuestionPaymentEntity.createNewEntity(questionPayment))
        return questionPayment
    }

    override fun countByUserId(userId: Long): Int {
        return questionPaymentJpaRepository.countByUserId(userId)
    }
}
